spec/syntax/atom_spec.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe 'Atom syntax' do
+  KEYWORDS = %w(
+    def
+    defp
+    defmodule
+    defprotocol
+    defimpl
+    defrecord
+    defrecordp
+    defmacro
+    defmacrop
+    defdelegate
+    defoverridable
+    defexception
+    defcallback
+    defstruct
+  )
+
+  it '`atom:` style keyword used as an atom' do
+    KEYWORDS.each do |kw|
```

Comment:
@kassio not sure what you think of doing things like this in tests. I tend to like them for increasing test coverage, but I'm willing to split this out if you'd rather